WIP

michaelwood opened the pull request:
Is this a long the right lines?

odscjames commented:
Calculating per feed looks good.

But I think it should be:

Passed & validated & not deleted / validated & not deleted

Deleted data never gets validated anyway and it seems like validated should be on the bottom otherwise someone's score is penalised if we just haven't processed their data yet and that's not fair.

I've just realised, if validated == 0 it would be nice if the result was N/A somehow?

michaelwood commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Status Page: Add source data validation stats](https://trello.com/c/gJ7uh6B6/65-status-page-add-source-data-validation-stats)

odscjames commented:
So we are showing

* validation passed %, which includes as a fail those not processed yet
* validation processed %

I would have thought it would be clearer to show the former as totally excluding those that are not processed. 

With this system, if you see "validation passed % == 30%" you have to check "validation processed %" to work out if you have problems with the data or it's just not checked yet.

If not processed yet is totally excluded, you can see  "validation passed % == 30%" and know straight away the data is bad.

But I don't know if there has been discussions with client about this I'm not aware of, so if there has do ignore this comment.

odscjames commented:
The code looks good, except are we sure we won't get a divide by zero error on publishers with no raw data? I'll approve code and if your happy with div/0 and the design comment above do merge.

michaelwood commented:
The div/0 will just cause the output to be null, which will be helpful in the ui to determine the difference between the states. I have fixed the issue of the validPercent including results which haven't been validated yet (and are therefore false) as I agree it would actually be confusing. Especially now that I added a validationDonePercent which will contextualise the percent.
